Lib/xml/etree/ElementTree.py

```diff
@@ -897,7 +897,7 @@ def _serialize_xml(write, elem, qnames, namespaces,
     tag = elem.tag
     text = elem.text
     if tag is Comment:
-        write("<!--%s-->" % text)
+        write("<!--%s-->" % _escape_cdata(text))
     elif tag is ProcessingInstruction:
         write("<?%s?>" % text)
```

Comment:
Please also include the text of processing instructions.

Reply:
It now escapes "?>" with "&gt;>".

Reply:
Sorry for this, I was wrong. PIs are not escaped. Please revert the commit where you added it.